packages/components/app/styles/components/code-block/index.scss

```diff
@@ -0,0 +1,219 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// CODE-BLOCK
+//
+
+
+// Note: "theme" contains just color variables and syntax highlighting styles
+@import "theme";
+
+// DIMENSIONS
+$code-block-line-numbers-width: 49px; // 3em = 49px
+$code-block-code-padding: 16px;
+
+// CODE-BLOCK PARENT/WRAPPER
+
+.hds-code-block {
+  position: relative;
+  overflow: hidden; // So header corners don't poke out from container
+  color: var(--code-block-color-foreground-primary);
+  font-family: var(--hds-typography-code-200);
+  background-color: var(--code-block-color-surface-primary);
+  border: 1px solid var(--code-block-color-border-strong);
+
+  pre,
+  code {
+    line-height: var(--token-typography-body-200-line-height); // = 1.4286 | 20px
+    white-space: pre;
+    text-align: left;
+    text-shadow: none;
+    word-wrap: normal;
+    word-break: normal;
+    word-spacing: normal;
+    tab-size: 4;
+    hyphens: none;
+
+    @media print { text-shadow: none; }
+  }
+}
+
+// CHILD COMPONENTS / ELEMENTS
+
+// Header (contains title & description)
+.hds-code-block__header {
+  display: flex;
+  flex-flow: column;
+  gap: 4px;
+  padding: 8px $code-block-code-padding;
+  background-color: var(--code-block-color-surface-faint);
+  border-bottom: 1px solid var(--code-block-color-border-primary);
+
+  &:empty {
+    display: none;
+  }
+}
+
+.hds-code-block__description {
+  color: var(--code-block-color-foreground-faint);
+
+  > a {
+    color: var(--code-block-color-foreground-action-default);
+  }
+}
+
+// Code
+.hds-code-block__code {
+  display: block;
+  margin: 0;
+  padding: $code-block-code-padding;
+  overflow: auto;
+}
+
+// CopyButton
+.hds-code-block__copy-button {
+  // Overriding default colors
+  color: var(--code-block-color-foreground-primary);
+  background-color: var(--code-block-color-surface-faint);
+  border: 1px solid var(--code-block-color-border-strong);
+
+  &:hover,
+  &.mock-hover {
+    background-color: var(--code-block-color-surface-primary);
+    border-color: var(--code-block-color-border-strong);
+  }
+
+  &:active,
+  &.mock-active {
+    background-color: var(--code-block-color-surface-interactive-active);
+  }
+
+  &:focus,
+  &.mock-focus,
+  &:focus-visible {
+    background-color: var(--code-block-color-surface-faint);
+  }
+
+  &.hds-copy-button--status-success {
+    .flight-icon {
+      color: var(--code-block-color-foreground-success);
+    }
+  }
+
+  &.hds-copy-button--status-error {
+    .flight-icon {
+      color: var(--code-block-color-foreground-critical);
+    }
+  }
+
+  .flight-icon { color: var(--code-block-color-foreground-primary); }
+}
+
+// VARIANTS
+
+// isRounded
+.hds-code-block--is-rounded {
+  border-radius: 6px;
+}
+
+// hasCopyButton
+.hds-code-block--has-copy-button {
+  .hds-code-block__copy-button {
```

Comment:
This can be moved inside the `.hds-code-block__copy-button` declaration, since the button is visible **only** if `@hasCopyButton` is true, so this double selector is redundant

Reply:
Refactored in 97f0f1ef9231f5b11f73f7c39074eb9b1cbcec6c. I think the main reason was to separate the subcomponent styles from the in-context positioning of the subcomponent within the component.